chain-impl-mockchain/src/tokens.rs

```diff
@@ -0,0 +1,208 @@
+use chain_core::mempack::{ReadBuf, ReadError, Readable};
+use cryptoxide::{blake2b::Blake2b, digest::Digest};
+use thiserror::Error;
+use typed_bytes::ByteBuilder;
+
+use std::convert::TryInto;
+
+pub const POLICY_HASH_SIZE: usize = 28;
+pub const TOKEN_NAME_MAX_BYTES: usize = 32;
+
+#[derive(Debug, Clone, Hash, PartialEq, Eq)]
+pub struct TokenIdentifier {
+    pub policy_hash: PolicyHash,
+    pub token_name: TokenName,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct PolicyHash([u8; POLICY_HASH_SIZE]);
```

Comment:
Maybe we could have a struct like `Blake2b224` (we already have `Blake2b256`) so that the hash type is also enforced

Reply:
Well, I do not see a particular reason to do this. We need to serialize and deserialize this type anyway. At this point there is no "hash of a particular type", just some bytes that may happen to be a valid hash.